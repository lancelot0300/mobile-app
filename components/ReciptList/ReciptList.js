import {
  ScrollView,
  Text,
  TouchableOpacity,
  FlatList,
  Image,
} from "react-native";
import React, { useState } from "react";
import drinkIcon from "../../assets/drink.png";


const ReciptList = ({ data, setDrink, setModalVisible }) => {
  return (
    <>
      {data && data.length !== 0 && (
        <FlatList
          data={data}
          renderItem={({ item }) => (
            <TouchableOpacity
              style={{
                backgroundColor: "rgba(255,255,255,0.05)",
                padding: 10,
                borderRadius: 10,
                margin: 10,
                borderColor: "#fff",
                height: 150,
                flex: 1,
                justifyContent: "center",
                alignItems: "center",
              }}
              onPress={() => {
                setDrink(item);
                setModalVisible(true);
              }}
            >
              <Image
                source={drinkIcon}
                style={{
                  width: 70,
                  height: 70,
                  alignSelf: "center",
                  marginTop: 10,
                  marginBottom: 20,
                }}
              />
              <Text style={{ color: "#fff", textAlign: "center" }}>
                {item.name}
              </Text>
            </TouchableOpacity>
          )}
          keyExtractor={(item, index) => item.name + index}
          numColumns={2}
          key={"_"}
          style={{ flex: 1, width: "100%", height: "100%" }}
        />
      )}
    </>
  );
};

export default ReciptList;
