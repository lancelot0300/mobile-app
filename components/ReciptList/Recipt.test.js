import React from "react";
import renderer from "react-test-renderer";
import { View } from "react-native";

import  ReciptList from "./ReciptList";

describe("<ReciptList />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ReciptList
          data={{
            drink1: {
              ingredients: {
                ingedient1: "true",
                ingedient2: "false",
                ingedient3: "true",
              },
              instructions: "instruction1",
              name: "margarita",
            },
          }}
          setDrink={() => {
            return;
          }}
          setModalVisible={() => {
            return;
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

    it("showing view", () => {
        const tree = renderer
        .create(<ReciptList
            data={{
              
            }}
            setDrink={() => {
              return;
            }}
            setModalVisible={() => {
              return;
            }}
          />
        ).toJSON();
        expect(tree.type).toBe('RCTScrollView');
    }
    );
});
