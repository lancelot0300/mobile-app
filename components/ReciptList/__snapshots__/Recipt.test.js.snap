// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReciptList /> renders correctly 1`] = `
<RCTScrollView
  data={
    {
      "drink1": {
        "ingredients": {
          "ingedient1": "true",
          "ingedient2": "false",
          "ingedient3": "true",
        },
        "instructions": "instruction1",
        "name": "margarita",
      },
    }
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    {
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
  viewabilityConfigCallbackPairs={[]}
>
  <View />
</RCTScrollView>
`;
